### PR TITLE
Working Proof Tests Using Proposals

### PR DESCRIPTION
**Pull request by nodlesh**
3 new Proof test scenario outlines for Proposal are included in this submission. A total of 5 new executable test scenarios.
T003-AIP10-RFC0037 - Present Proof where the prover has proposed the presentation of proof in response to a presentation request and is acknowledged.
T003.1-AIP10-RFC0037 - Present Proof where the prover has proposed the presentation of proof from a different credential in response to a presentation request and is acknowledged.
T006-AIP10-RFC0037 - Present Proof where the prover starts with a proposed the presentation of proof and is acknowledged.

### CODE REVIEW

File: TEST_DEV_GUIDE.md [hunk 1]
```diff
@@ -103,15 +102,15 @@ Defining test steps require using and extending the commands and operations to b
 
 ## Implementing Test Steps
 
-Follow standard best practices for implementing test steps in Behave, writing the test steps as if the feature is fully supported and then adding code at all levels to support the new test. The process is something like this:
+Follow standard best practices for implementing test steps in Behave, writing the test steps as if the feature is fully supported and then adding code at all levels to supprt the new test. The process is something like this:
```
Comment (swcurran) [inline TEST_DEV_GUIDE.md]: Typo

File: TEST_DEV_GUIDE.md [hunk 2]
```diff
@@ -103,15 +102,15 @@ Defining test steps require using and extending the commands and operations to b
...
 
 - Study the RFC and define the tests needed to cover at least the happy path requirements of the RFC
   - Raise as issues in the [Aries RFCs](https://github.com/hyperledger/aries-rfcs) repo any deficiencies in the RFC
 - Define Gherkin tests using the existing persona
   - Add new persona only if really, really needed
 - Execute the new tests, grabbing the skeleton code for the new, unimplemented steps from the behave output, and adding them to the `steps` Python code
 - Implement the steps code, defining as needed commands and operations to be added to the backchannel interface
-- Update the Google Sheet and OpenAPI spec to add the new commands and operations and regenerate the backchannel operations data file (see [subsection](#defining-backchannel-operations) above)
+- Update the Google Sheet to add the new commands and operations and regenerate the backchannel operations data file (see [subsection](#defining-backchannel-operations) above)
```
Comment (swcurran) [inline TEST_DEV_GUIDE.md]: This seems wrong -- looks like things are being undone?  We need to update the OpenAPI Spec.

Perhaps a restore bug?

File: TEST_DEV_GUIDE.md [hunk 3]
```diff
@@ -123,5 +122,4 @@ See the [README](../aries-agent-test-harness/aries-backchannels/README.md) in th
 
 ## Diving Deeper
 - [Accessing Connection ID in Test Code](ACCESS-CONNECTION-IDS.md)
-- [Configuring Tests with Credential Types and Proofs](CONFIGURE-CRED-TYPES.md)
-- [Debugging a Backchannel Running Inside a Docker Contaiener](DEBUGGING.md)
```
Comment (swcurran) [inline TEST_DEV_GUIDE.md]: Same?

**Review by swcurran — CHANGES_REQUESTED** [review]
@nodlesh -- looks like you may have some files that shouldn't be changed in this?

**Comment by swcurran** [comment]
It ran well, so there is that :-)

File: TEST_DEV_GUIDE.md (hunk 3 above)
Comment (nodlesh) [reply TEST_DEV_GUIDE.md]: hmmm, that is strange. I have no idea what happened here.

**Comment by TimoGlastra** [comment]
This PR is behind master and doesn't have the documentation changes I made in #92 yet

File: TEST_DEV_GUIDE.md (hunk 1 above)
Comment (nodlesh) [reply TEST_DEV_GUIDE.md]: File properly merged in new commit.

File: TEST_DEV_GUIDE.md (hunk 2 above)
Comment (nodlesh) [reply TEST_DEV_GUIDE.md]: File properly merged in new commit.

File: TEST_DEV_GUIDE.md (hunk 3 above)
Comment (nodlesh) [reply TEST_DEV_GUIDE.md]: File properly merged in new commit.

**Review by swcurran — APPROVED** [review]
Ran through the changes and executed the suite -- looks good.

**Pull request merged**